Elastic path/source (#5251)
segmentio/segment-docs: src/connections/sources/catalog/cloud-apps/elastic-path/index.md

| `userId`                | Customer ID of Elastic Path Composable Commerce                               |
| `traits`                | Free-form dictionary of traits of the user, like email or                     |
| `traits.name`           | Customer name                                                                 |
| `traits.email`          | Customer email                                                                |

**Note**: You can add any custom fields as well as part of the traits e.g. age, gender, title, phone etc.

### Order events

The following payload contains the properties for the following events:

- `Checkout Started`
- `Order Completed`
- `Order Updated`
- `Order Cancelled`
- `Order Refunded`

| Property Name                                       | Description                                                                                                                                           |
| --------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `userId`                                | If registered user place order then the value of userId will be the Customer ID otherwise it will be order ID. You can change this as per your requirement.                                                                                                       |
> | `event`                                 | Event name e.g. `Checkout Started`, `Order Completed` etc.             |
> | `properties`                            | This will hold information about the order                             |
> | `properties.order_id`                   | Elastic Path Order ID                                                  |
> | `properties.value`                      | Revenue ($) with discounts and coupons added in. For better flexibility and total control over tracking, we let you decide how to calculate how coupons and discounts are applied. Usually this is required for `Checkout Started` event                                                                              |
> | `properties.total`                      | Revenue ($) with discounts and coupons added in. For better flexibility and total control over tracking, we let you decide how to calculate how coupons and discounts are applied. Usually this is required for all the events apart from `Checkout Started` event                                                    |
> | `properties.revenue`                    | Revenue ($) associated with the transaction (excluding shipping and tax). Usually this is required for all the events apart from `Checkout Started` event                                    |
> | `properties.tax`                        | Total tax associated with the transaction                              |
> | `properties.discount`                   | Total discount associated with the transaction                         |
> | `properties.shipping`                   | Shipping cost associated with the transaction                          |
> | `properties.currency`                   | Currency code associated with the transaction e.g. GBP, USD etc.       |
> | `properties.products`                   | List of products in the order                                          |
> | `properties.products.$.product_id`      | Product ID                                                             |
> | `properties.products.$.name`            | Product Name                                                           |
> | `properties.products.$.sku`             | Product SKU                                                            |
> | `properties.products.$.price`           | Product Unit Price                                                     |
> | `properties.products.$.quantity`        | Quantity of a product                                                  |

Reviewer: ```suggestion
| `userId`                                | If the registered user places an order, then the value of the userId is the Customer ID. Otherwise, it will be the order ID. You can change this as per your requirement.                                                                                                       |
| `event`                                 | The event name, for example,  `Checkout Started`, `Order Completed`.             |
| `properties`                            | This holds information about the order.                             |
| `properties.order_id`                   | The Elastic Path Order ID.                                                  |
| `properties.value`                      | The revenue ($) with discounts and coupons added in. For better flexibility and total control over tracking, Elastic Path lets you decide how to calculate how coupons and discounts are applied. This is usually required for the `Checkout Started` event.                                                                              |
| `properties.total`                      | The revenue ($) with discounts and coupons added in. For better flexibility and total control over tracking, Elastic Path lets you decide how to calculate how coupons and discounts are applied. This is usually required for all the events apart from the `Checkout Started` event.                                                    |
| `properties.revenue`                    | The revenue ($) associated with the transaction (excluding shipping and tax). This is usually required for all the events apart from the `Checkout Started` event.                                    |
| `properties.tax`                        | The total tax associated with the transaction.                              |
| `properties.discount`                   | The total discount associated with the transaction.                         |
| `properties.shipping`                   | The shipping cost associated with the transaction.                          |
| `properties.currency`                   | The currency code associated with the transaction. For example, GBP, USD.       |
| `properties.products`                   | The list of products in order.                                          |
| `properties.products.$.product_id`      | The product ID.                                                             |
| `properties.products.$.name`            | The product name.                                                           |
| `properties.products.$.sku`             | The product SKU.                                                            |
| `properties.products.$.price`           | The product unit price.                                                     |
| `properties.products.$.quantity`        | The quantity of a product.                                                  |
```

> Note: We are using properties of these events from [eCommerce Spec](https://segment-docs.netlify.app/docs/connections/spec/ecommerce/v2){:target="_blank"} defined by Segment doc. Add more properties as per this documentation based on your business requirement.

Reviewer: ```suggestion
Note: Elastic path is using properties of these events from the [eCommerce Spec](https://segment-docs.netlify.app/docs/connections/spec/ecommerce/v2){:target="_blank"} defined by the Segment doc. Add more properties as per this documentation based on your business requirement.
```

## Add destinations

> Now that your Source is configured, you can connect it with Destinations. Log into your downstream tools and check to see that your events appear as expected, and that they contain all of the properties you expect. If your events and properties don’t appear, check the Event Delivery tool, and refer to the Destination docs for each tool for troubleshooting. If there are any issues with how the events are arriving to Segment, contact the [Elastic Path's Support team](https://support.elasticpath.com/hc/en-us){:target="_blank"}.

Reviewer: ```suggestion
Now that you've configured your source, you can connect it with destinations. Log in to your downstream tools and check to see that your events appear as expected, and that they contain all of the properties you expect. If your events and properties don’t appear, check the Event Delivery tool, and refer to the Destination docs for each tool for troubleshooting. If there are any issues with how the events arrive to Segment, contact the [Elastic Path Support team](https://support.elasticpath.com/hc/en-us){:target="_blank"}.
```
